Automattic/wp-calypso
# chore: run `remove-overloads` codemod on query files

Part of #84338 

## Proposed Changes

This PR runs the [`remove-overloads` codemod](https://tanstack.com/query/v5/docs/react/guides/migrating-to-v5#codemod) on all files in `/packages`, `/apps` and `/client`. RQ v5 only supports object syntax for most of the query methods which is why we need those changes in order to be able to update.

Please note, there are a few occurrences left after this PR because the codemod only touches methods which are given literal arrays as a parameter. The codemod can't know whether a variable passed to e.g. `cancelQueries( myVar );` contains a valid object or an array. We have to update these manually.

## Testing Instructions

For one verify Calypso builds well, all tests are passing, there are no linting issues. Smoke testing Calypso won't hurt. Also, verify there are only changes which change query methods to using object syntax (compare the relevant part in the [migration guide for RQ v5](https://tanstack.com/query/v5/docs/react/guides/migrating-to-v5#supports-a-single-signature-one-object)).
